front/lib/api/assistant/conversation.ts

```diff
@@ -175,6 +183,119 @@ export async function* postUserMessage(
   | AgentMessageTokensEvent
   | AgentMessageSuccessEvent
 > {
+  const user = auth.user();
+
+  let userMessageEvent: UserMessageNewEvent | null = null;
+  const agentMessageEvents: AgentMessageNewEvent[] = [];
+
+  await front_sequelize.transaction(async (t) => {
+    let nextMessageRank =
+      ((await AssistantMessage.max<number | null, AssistantMessage>("rank", {
+        where: {
+          assistantConversationId: conversation.id,
+        },
+        transaction: t,
+      })) ?? -1) + 1;
+    const userMessage = await AssistantMessage.create(
+      {
+        sId: generateModelSId(),
+        rank: nextMessageRank++,
+        assistantConversationId: conversation.id,
+        parentId: null,
+        assistantUserMessageId: (
+          await AssistantUserMessage.create(
+            {
+              message: message,
+              userContextUsername: context.username,
+              userContextTimezone: context.timezone,
+              userContextFullName: context.fullName,
+              userContextEmail: context.email,
+              userContextProfilePictureUrl: context.profilePictureUrl,
+              userId: user ? user.id : null,
+            },
+            { transaction: t }
+          )
+        ).id,
+      },
+      {
+        transaction: t,
+      }
+    );
+    userMessageEvent = {
+      type: "user_message_new",
+      message: {
+        id: userMessage.id,
+        sId: userMessage.sId,
+        type: "user_message",
+        visibility: "visible",
+        version: 0,
+        parentMessageId: null,
+        user: user,
+        mentions: mentions,
+        message: message,
+        context: context,
+      },
+    };
+
+    // for each assistant mention, create an "empty" agent message
+    for (const m of mentions) {
+      if (isAssistantAgentMention(m)) {
+        const agentMessage = await AssistantMessage.create(
+          {
+            sId: generateModelSId(),
+            rank: nextMessageRank++,
+            assistantConversationId: conversation.id,
+            parentId: userMessage.id,
+            assistantAgentMessageId: (
+              await AssistantAgentMessage.create({}, { transaction: t })
+            ).id,
+          },
+          {
+            transaction: t,
+          }
+        );
+
+        agentMessageEvents.push({
+          type: "agent_message_new",
+          created: Date.now(),
+          configurationId: m.configurationId,
+          message: {
```

Comment:
You will need that message later on so you can directly store it in its own variable before passing it in the event?

Reply:
done